src/user/user.service.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable } from '@nestjs/common';
+import { CreateUserDto } from './dto/create-user.dto';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { hash } from 'argon2';
+import {
+  BadRequestException,
+  NotFoundException,
+} from '@nestjs/common/exceptions';
+import { Prisma } from '@prisma/client';
+
+@Injectable()
+export class UserService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(createUserDto: CreateUserDto) {
+    const oldUser = await this.prisma.user.findUnique({
+      where: { email: createUserDto.email },
+    });
+
+    if (oldUser) {
+      throw new BadRequestException('User already exist');
+    }
+
+    const user = await this.prisma.user.create({
+      data: {
+        email: createUserDto.email,
+        name: createUserDto.name,
+        password: await hash(createUserDto.password),
+      },
+    });
+
+    return {
+      id: user.id,
+      name: user.name,
+      email: user.email,
+    };
```

Comment:
You can use something like `omit` function from `lodash`/`radash`:
```ts
return omit(user, ['password']);
```

Or you can use object destruction:
```ts
const { password, ...userWithoutPassword } = user;

return userWithoutPassword;
```